labelCloud/io/segmentations/base.py

```diff
@@ -0,0 +1,58 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+import json
+from typing import Dict, Tuple, Type
+import numpy.typing as npt
+import numpy as np
+from ...utils.singleton import SingletonABCMeta
+
+
+class BaseSegmentationHandler(object, metaclass=SingletonABCMeta):
+    EXTENSIONS = set()  # should be set in subclasses
+
+    def __init__(self, label_definition_path: Path) -> None:
+        self.read_label_definition(label_definition_path)
+
+    def read_label_definition(self, label_definition_path: Path) -> None:
+        with open(label_definition_path, "r") as f:
+            self.label_definition: Dict[str, int] = json.loads(f.read())
+            assert len(self.label_definition) > 0
```

Comment:
Instead of failing here a warning would be more user friendly. ;)

... but fine for now!